verification/VerifyImputation.wdl

```diff
@@ -0,0 +1,162 @@
+## authorized to run all programs before running this script. Please see the docker
+## page at https://hub.docker.com/r/broadinstitute/genomes-in-the-cloud/ for detailed
+## licensing information pertaining to the included programs.
+
+workflow VerifyImputation {
+  input {
+    File haplotype_database
+    Boolean split_output_to_single_sample
+
+    Array[File] truth_metrics
+    Array[File] test_metrics
+
+    File truth_vcf
+    File test_vcf
+    File test_vcf_index
+
+    File? input_multi_sample_vcf
+    File? input_multi_sample_vcf_index
+    Array[File]? input_single_sample_vcfs
+    Array[File]? input_single_sample_vcfs_indices
+
+    Array[File]? single_sample_truth_vcf
+    Array[File]? single_sample_test_vcf
+    Array[File]? single_sample_test_vcf_indices
+  }
+
+  scatter (idx in range(length(truth_metrics))) {
+    call CompareImputationMetrics {
+      input:
+        test_metrics = test_metrics[idx],
+        truth_metrics = truth_metrics[idx]
+    }
+  }
+
+  call Tasks.CompareVcfs as CompareOutputVcfs {
+    input:
+      file1 = truth_vcf,
+      file2 = test_vcf
+  }
+
+  if (defined(single_sample_truth_vcf)) {
+    scatter (idx in range(length(select_first([single_sample_truth_vcf])))) {
+      call Tasks.CompareVcfs as CompareSingleSampleOutputVcfs {
+        input:
+          file1 = select_first([single_sample_test_vcf])[idx],
+          file2 = select_first([single_sample_truth_vcf])[idx]
+      }
+    }
+  }
+
+  call CrosscheckFingerprints {
+    input:
+      firstInputs = if (defined(input_multi_sample_vcf)) then select_all([input_multi_sample_vcf]) else select_first([input_single_sample_vcfs]),
+      firstInputIndices = if (defined(input_multi_sample_vcf_index)) then select_all([input_multi_sample_vcf_index]) else select_first([input_single_sample_vcfs_indices]),
+      secondInputs = [test_vcf],
+      secondInputIndices = [test_vcf_index],
+      haplotypeDatabase = haplotype_database
+  }
+
+  if (split_output_to_single_sample) {
+    call ImputationTasks.SplitMultiSampleVcf {
+      input:
+        multiSampleVcf = test_vcf
+    }
+
+    call CrosscheckFingerprints as CrosscheckFingerprintsSplit {
+      input:
+        firstInputs = if (defined(input_multi_sample_vcf)) then select_all([input_multi_sample_vcf]) else select_first([input_single_sample_vcfs]),
+        firstInputIndices = if (defined(input_multi_sample_vcf_index)) then select_all([input_multi_sample_vcf_index]) else select_first([input_single_sample_vcfs_indices]),
+        secondInputs = SplitMultiSampleVcf.single_sample_vcfs,
+        secondInputIndices = SplitMultiSampleVcf.single_sample_vcf_indices,
+        haplotypeDatabase = haplotype_database
+    }
+  }
+
+  output {
+  }
+  meta {
+    allowNestedInputs: true
+  }
+}
+
+task CompareImputationMetrics {
+  input {
+    File test_metrics
+    File truth_metrics
+  }
+  command <<<
+  set -eo pipefail
+  diff "~{test_metrics}" "~{truth_metrics}"
```

Comment:
Did the CompareMetrics task not work?  Are these picard metrics?

Reply:
If i remember correctly, the metrics output by the pipeline were failing because the header had "illegal characters", but we inspected the files and couldn't figure out the issue. Since the metrics files are so small, we went ahead and just diffed them. @jessicaway am I forgetting anything?

Reply:
update -- these aren't picard metrics, so that is the main reason

Reply:
👍